bot.js

```diff
@@ -134,71 +134,77 @@ class Bot {
             return -1;
         }
     }
-    log() { /* {{{ */
-        let arr = ["[" + this.pid() + "]"];
-        for (let i=0; i < arguments.length; ++i) {
+    log() {
+        const arr = [ `[${this.pid()}]` ];
+        for (let i = 0; i < arguments.length; ++i) {
             arr.push(arguments[i]);
```

Comment:
1. Why this extra spaces at ```[ `[${this.pid()}]` ];```?
2. Do we need to string interpolate a single item? Also possible to do `this.pid().toString()`, but i think it will work as number too.
3. Why iterate an array to push items to a new array? Could just do `[this.pid(), ...arguments]`
4. Do we really need `apply` on `console.log.apply`?
5. Actually, we can pass it to the console function right away too right? `console.log(this.pid(), ...arguments)`?

(do test my above suggestion.)

Edit: this same comment applies to the error and verbose functions right below this comment.

Reply:
i know nothing about how this code works, but i'll open it in an issue to keep this refactoring in mind and test it :)

Reply:
kept in #305